ethereum/core/src/test/java/tech/pegasys/pantheon/ethereum/chain/GenesisStateTest.java

```diff
@@ -85,6 +85,19 @@ public void createFromJsonNoAllocs() throws Exception {
     assertThat(header.getParentHash()).isEqualTo(Hash.ZERO);
   }
 
+  @Test
+  public void createFromJsonWithContract() throws Exception {
```

Comment:
I might remove this test and edit `createFromJsonWithAllocs` but i left it here in order to prove that it passes the case described in issue.

Reply:
Yeah it could go either way but I think it's worth having a test that covers this more complex case specifically.  I've added a few assertions for the contract account's code and storage for good measure.